Wrappers/Python/ccpi/optimisation/algorithms/GradientDescent.py

```diff
@@ -65,29 +71,98 @@ def set_up(self, x_init, objective_function, rate):
             
         self.x = x_init.copy()
         self.objective_function = objective_function
-        self.rate = rate
+        
 
-        self.loss.append(objective_function(x_init))
+        if rate is None:
+            self.k = 0
+            self.update_rate = True
+            self.x_armijo = x_init.copy()
+            # self.rate = self.armijo_rule() * 2
+            # print (self.rate)
+        else:
+            self.rate = rate
```

Comment:
Can we please change the name of `rate` to `stepsize` or similar, because that is what it is. Rate is ambiguous and similar to "convergence rate", which is different from the step size.